From e290dcf25044b3b41406f50065812c69a29d9e64 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Thu, 17 Dec 2020 18:01:40 +0200
Subject: [PATCH 27/27] lx2160a: set dpaa mac registration as weak, and clear
 build warning msgs

1. Set fsl_mc_ldpaa_init so that a board can register the dpmacs in it's
own order
2. Cleanup build warning messages which is triggered when using
CONFIG_I2C_EEPROM required for sys_eeprom feature

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 drivers/net/fsl-mc/mc.c          | 2 +-
 include/configs/lx2160a_common.h | 5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/net/fsl-mc/mc.c b/drivers/net/fsl-mc/mc.c
index f7602493..2a0ec4a8 100644
--- a/drivers/net/fsl-mc/mc.c
+++ b/drivers/net/fsl-mc/mc.c
@@ -915,7 +915,7 @@ unsigned long mc_get_dram_block_size(void)
 	return dram_block_size;
 }
 
-int fsl_mc_ldpaa_init(bd_t *bis)
+__weak int fsl_mc_ldpaa_init(bd_t *bis)
 {
 	int i;
 
diff --git a/include/configs/lx2160a_common.h b/include/configs/lx2160a_common.h
index 288f0e14..c57b9ce2 100644
--- a/include/configs/lx2160a_common.h
+++ b/include/configs/lx2160a_common.h
@@ -119,6 +119,11 @@
 #define CONFIG_SYS_I2C_RTC_ADDR		0x51  /* Channel 3*/
 
 /* EEPROM */
+#undef CONFIG_SYS_EEPROM_BUS_NUM
+#undef CONFIG_SYS_I2C_EEPROM_ADDR
+#undef CONFIG_SYS_I2C_EEPROM_ADDR_LEN
+#undef CONFIG_SYS_EEPROM_PAGE_WRITE_BITS
+#undef CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS
 #define CONFIG_ID_EEPROM
 #define CONFIG_SYS_I2C_EEPROM_NXID
 #define CONFIG_SYS_EEPROM_BUS_NUM		0
-- 
2.25.1

